src/secp256k1/src/field_10x26_impl.h

```diff
@@ -809,11 +811,20 @@ SECP256K1_INLINE static void secp256k1_fe_sqr_inner(uint32_t *r, const uint32_t
      *  Note that [x 0 0 0 0 0 0 0 0 0 0] = [x*R1 x*R0].
      */
 
+#ifdef SECP256_PVS_FIX
     d  = ( (uint64_t)a[0]*2 ) * a[9]
        + ( (uint64_t)a[1]*2 ) * a[8]
        + ( (uint64_t)a[2]*2 ) * a[7]
        + ( (uint64_t)a[3]*2 ) * a[6]
        + ( (uint64_t)a[4]*2 ) * a[5];
+#else // !SECP256_PVS_FIX
```

Comment:
all new code analyzers warn about c-style casts, it's very difficult to replace all places, but if we touch some code - it make sense to change c-style cast to static_cast<uint64_t>()...

Reply:
Will fix

Reply:
Currently this library is compiled with a C compiler, so we should keep it here.

Reply:
My bad, yes... secp256k1 is indeed pure C, so yes..this is the only option there